docs/core-api/BLOCK.md

```diff
@@ -96,21 +96,6 @@ const decoder = new TextDecoder()
 
 const block = await ipfs.block.put(buf)
 
-console.log(decoder.decode(block.data))
-// Logs:
-// a serialized object
-console.log(block.cid.toString())
-// Logs:
-// the CID of the object
-
-// With custom format and hashtype through CID
-import { CID } from 'multiformats/cid'
-import * as dagPB from '@ipld/dag-pb'
-const buf = new TextEncoder().encode('another serialized object')
-const cid = CID.createV1(dagPB.code, multihash)
-
-const block = await ipfs.block.put(blob, cid)
```

Comment:
Couldn't find evidence of this being possible. Both the TypeScript typings and the implementation do not support a `CID` as parameter.

https://github.com/ipfs/js-ipfs/blob/548cfb215e8afa8dca64d3c38fa13feecbe577e2/packages/ipfs-core-types/src/block/index.ts#L50

https://github.com/ipfs/js-ipfs/blob/548cfb215e8afa8dca64d3c38fa13feecbe577e2/packages/ipfs-core/src/components/block/put.js#L20

Reply:
`CID` used to be supported as a parameter, but it's not safe as you could store the wrong block for a CID so it was removed.